### ISSUE TITLE
Microsoft .NET error when trying to log in a DropBox account

### ISSUE THREAD

**Issue by Klemet** [Kyrodan/KeeAnywhere#183]
Hello !

I can't log into a DropBox account in one of my computers. I've tried everything (restricted mode or not), but it always come back to this error : 

> See the end of this message for details on invoking 
> just-in-time (JIT) debugging instead of this dialog box.

> ************** Exception Text **************
> System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.
> File name: 'Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
>    at Dropbox.Api.Stone.JsonWriter.Write[T](T encodable, IEncoder`1 encoder, Boolean escapeNonAscii)
>    at Dropbox.Api.DropboxRequestHandler.<Dropbox.Api.Stone.ITransport.SendRpcRequestAsync>d__5`3.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at KeeAnywhere.StorageProviders.Dropbox.DropboxStorageConfigurator.<CreateAccount>d__1.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at KeeAnywhere.StorageProviders.StorageService.<CreateAccount>d__9.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at KeeAnywhere.UIService.<CreateOrUpdateAccount>d__5.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at KeeAnywhere.Forms.SettingsForm.<OnAccountAdd>d__a.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
> 
> WRN: Assembly binding logging is turned OFF.
> To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
> Note: There is some performance penalty associated with assembly bind failure logging.
> To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
> 
> 
> 
> Loaded Assemblies 
>
> mscorlib
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3416.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll

> KeePass
>     Assembly Version: 2.42.1.23464
>     Win32 Version: 2.42.1.0
>     CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/KeePass.exe

> System
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3416.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll

> System.Windows.Forms
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3324.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll

> System.Drawing
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll

> System.Configuration
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3324.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll

> System.Core
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3429.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll

> System.Xml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll

> WebSiteAdvantageKeePassFirefoxImporter
>     Assembly Version: 2.28.0.0
>     Win32 Version: 2.28.0.0
>     CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/WebSiteAdvantageKeePassFirefoxImporter.DLL

> KeeAnywhere
>     Assembly Version: 1.5.1.0
>     Win32 Version: 1.5.1.0
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/6HEgCyiOBzNxwsMf6T54/KeeAnywhere.dll

> System.Net.Http.Primitives
>     Assembly Version: 4.2.29.0
>     Win32 Version: 4.2.29.0
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/6HEgCyiOBzNxwsMf6T54/System.Net.Http.Primitives.DLL

> Newtonsoft.Json
>     Assembly Version: 6.0.0.0
>     Win32 Version: 6.0.8.18111
>     CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/Newtonsoft.Json.DLL

> Microsoft.Graph.Core
>     Assembly Version: 1.9.0.0
>     Win32 Version: 1.9.0.0
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/6HEgCyiOBzNxwsMf6T54/Microsoft.Graph.Core.DLL

> Newtonsoft.Json
>     Assembly Version: 10.0.0.0
>     Win32 Version: 10.0.3.21018
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/6HEgCyiOBzNxwsMf6T54/Newtonsoft.Json.DLL

> System.Numerics
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll

> System.Runtime.Serialization
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3440.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll

> System.Data
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3260.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll

> CommonMark
>     Assembly Version: 0.1.0.0
>     Win32 Version: 0.1.0.0
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/6HEgCyiOBzNxwsMf6T54/CommonMark.DLL

> Microsoft.mshtml
>     Assembly Version: 7.0.3300.0
>     Win32 Version: 7.0.3300.0
>     CodeBase: file:///C:/WINDOWS/assembly/GAC/Microsoft.mshtml/7.0.3300.0__b03f5f7f11d50a3a/Microsoft.mshtml.dll

> KeeCloud
>     Assembly Version: 1.2.1.11
>     Win32 Version: 1.2.1.11
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/1nBVnxzCKD1475FzC2it/KeeCloud.dll

> KPDataSave
>     Assembly Version: 0.0.6.0
>     Win32 Version: 0.0.6.0
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/LzOek5m5yZKKoOJ7Od2j/KPDataSave.dll

> RestSharp
>     Assembly Version: 104.2.0.0
>     Win32 Version: 104.2.0.0
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/LzOek5m5yZKKoOJ7Od2j/RestSharp.DLL

> System.Security
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Security/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Security.dll

> KeePass.XmlSerializers
>     Assembly Version: 2.42.1.23464
>     Win32 Version: 2.42.1.23464
>     CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/KeePass.XmlSerializers.DLL

> Dropbox.Api
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.0.6718
>     CodeBase: file:///C:/Users/lenovo/AppData/Local/KeePass/PluginCache/6HEgCyiOBzNxwsMf6T54/Dropbox.Api.DLL

> System.Net.Http
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll

> Anonymously Hosted DynamicMethods Assembly
>     Assembly Version: 0.0.0.0
>     Win32 Version: 4.7.3416.0 built by: NET472REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll

Having every requirement for the plugin, I don't know what's wrong. I hope that this won't be a waste of time !

**Comment by Klemet** [comment]
Problem was resolved by replacing the Newtonsoft.Json.dll in the main folder of keypass (which was version 6.0.0.0 as indicated in the error message) with the Newtonsoft.Json.dll in the plugin cache folder of keypass (which was at version 10.0.0.0).

It looks like KeyPass installed the wrong version of the .dll, maybe. It's strange, since I did multiple re-install in trying to fix the problem. Because of that, it looks like the code was trying to load the old Newtonsoft.Json.dll (maybe by taking the one in the KeePass main folder by default), which did not have the functions needed to accomodate the plugin.
